tests/unit/resources/test_compensation.py

```diff
@@ -98,3 +108,143 @@ def test_create_from_spill_string(spillstring):
         "Qdot655-A",
         "Qdot705-A",
     ]
+
+
+@responses.activate
+def test_apply_scale_errors_for_nonmatching_channels(
+    client, ENDPOINT_BASE, compensation, fcs_file
+):
+    events_body = open("tests/data/Acea - Novocyte.fcs", "rb")
+    responses.add(
+        responses.GET,
+        f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs",
+        body=events_body,
+    )
+    events = parse_fcs_file(client.download_fcs_file(EXP_ID, fcs_file._id))
+    fcs_file.events = events
+
+    with pytest.raises(IndexError):
+        compensation.apply(fcs_file)
+
+
+@responses.activate
+def test_apply_compensation_to_fcs_file(compensation, fcs_file):
```

Comment:
This test can probably be removed. I don't think there are any branches in the code that this covers vs. `test_apply_scale_compensates_values`.